src/Scripts/fov.gd

```diff
@@ -0,0 +1,78 @@
+extends Node
+
+# FOV with recursive shadow casting, translated from python
+# see: http://www.roguebasin.com/index.php?title=Python_shadowcasting_implementation
+
+# Multipliers for transforming coordinates to other octants:
+const MULT = [
+	[1,  0,  0, -1, -1,  0,  0,  1],
+	[0,  1, -1,  0,  0, -1,  1,  0],
+	[0,  1,  1,  0,  0, -1, -1,  0],
+	[1,  0,  0,  1, -1,  0,  0, -1]
+]
+
+const BLOCKED_INDEXES = [1, 4]
+
+func blocked(tilemap, x, y):
+	return (x < 0 or y < 0 \
+			or x >= Constants.TILES_WIDE or y >= Constants.TILES_HIGH \
+			or tilemap.get_cell(x, y) in BLOCKED_INDEXES)
+
+func set_lit(fov, x, y):
+	if x >= 0 and x < Constants.TILES_WIDE and y >= 0 and y < Constants.TILES_HIGH:
+		fov[Vector2(x, y)] = true
+
+func _cast_light(tilemap, from_position, row, start, end, radius, xx, xy, yx, yy, id, fov):
+	# Recursive lightcasting function
+	if start < end:
+		return
+	
+	var radius_squared = radius*radius
+	var new_start
+	for j in range(row, radius+1):
+		var dx = -j-1
+		var dy = -j
+		var blocked = false
+		while dx <= 0:
+			dx += 1
+			# Translate the dx, dy coordinates into map coordinates
+			var X = from_position.x + dx * xx + dy * xy
```

Comment:
Naming some variables `x` and others `X` is probably going to cause confusion/chaos if someone ever looks into this code ro has to debug it